zerver/lib/actions.py

```diff
@@ -858,6 +858,13 @@ def do_change_realm_subdomain(realm: Realm, new_subdomain: str) -> None:
     realm.string_id = new_subdomain
     realm.save(update_fields=["string_id"])
 
+def do_add_deactivated_redirect(realm: Realm, redirect_url: str) -> None:
```

Comment:
Why did you choose this approach over making `redirect_url` an optional argument to `do_deactivate_realm`? Wouldn't that be a better approach?

Reply:
I saw that `do_deactivate_realm` is being used at quite a lot of other places. Plus I saw that we have functions like `do_change_realm_subdomain` ... So, I thought I should add another method for adding the `redirect_url` ... 

But, I understand your point as well, since we are using `deactivate_realm` command to add the `redirect_url` , we should add it in the `do_deactivate_realm` . Will make the change and push it! Thanks :)

Reply:
Now that I think about it a little more, it may be good to have a separate function if we are actually allowing changing the deactivate_realm field even after the realm has been deactivated.